integration/mediation-tests/tests-other/src/test/java/org/wso2/carbon/esb/connector/test/MediationLibraryServiceTestCase.java

```diff
@@ -0,0 +1,286 @@
+*
+*  http://www.apache.org/licenses/LICENSE-2.0
+*
+*  Unless required by applicable law or agreed to in writing,
+*  software distributed under the License is distributed on an
+*  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+*  KIND, either express or implied.  See the License for the
+*  specific language governing permissions and limitations
+*  under the License.
+*/
+package org.wso2.carbon.esb.connector.test;
+
+import org.apache.axis2.AxisFault;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.automation.test.utils.http.client.HttpRequestUtil;
+import org.wso2.carbon.automation.test.utils.http.client.HttpResponse;
+import org.wso2.carbon.mediation.library.stub.types.carbon.LibraryInfo;
+import org.wso2.esb.integration.common.utils.ESBIntegrationTest;
+import org.wso2.esb.integration.common.utils.common.ServerConfigurationManager;
+
+import java.io.File;
+import java.net.URL;
+
+/**
+ * Test the functionality of MediationLibraryServiceComponent
+ */
+public class MediationLibraryServiceTestCase extends ESBIntegrationTest {
+    private static final String HELLO_CONNECTOR_ZIP = "hello-connector-1.0.0-SNAPSHOT.zip";
+    private static final String HELLO_CONNECTOR_LIB_QNAME = "{org.wso2.carbon.connector}hello";
+    private static final String HELLO_LIB_NAME = "hello";
+
+    private static final String AMAZON_CONNECTOR_ZIP = "amazons3-connector-1.0.4.zip";
+    private static final String AMAZON_CONNECTOR_LIB_QNAME = "{org.wso2.carbon.connector}amazons3";
+    private static final String AMAZON_LIB_NAME = "amazons3";
+
+    private static final String TEMP_CONNECTOR_LIB_QNAME = "{org.wso2.carbon.connector}temp";
+    private static final String TEMP_LIB_NAME = "temp";
+
+    private static final String FAULTY_CONNECTOR_ZIP_NAME = "helloworld-connector-1.0.5-SNAPSHOT.zip";
+    private static final String FAULTY_CONNECTOR_LIB_QNAME = "{org.wso2.carbon.connector}helloworld";
+    private static final String FAULTY_LIB_NAME = "helloworld";
+
+    private static final String PACKAGE_NAME = "org.wso2.carbon.connector";
+    private static final String ENABLED = "enabled";
+    private static final String CONNECTOR = "connector";
+    private static final String INVALID_CONNECTOR = "invalid-connector";
+
+    private ServerConfigurationManager serverConfigurationManager;
+    private String resourcePath;
+
+    /**
+     * Deploy and enable the hello connector.
+     *
+     * @throws Exception
+     */
+    @BeforeClass(alwaysRun = true)
+    public void setEnvironment() throws Exception {
+
+        super.init();
+        serverConfigurationManager = new ServerConfigurationManager(context);
+        resourcePath = getESBResourceLocation().replace("//", "/") + File.separator + CONNECTOR;
+
+        //upload connector
+        uploadConnector(resourcePath, HELLO_CONNECTOR_ZIP);
+        Thread.sleep(20000);
```

Comment:
Shall we improve the sleep to wait until the connector is available in the resource path?

Reply:
Updated as suggested.